workload_generator/rest_client.py

```diff
@@ -0,0 +1,69 @@
+import requests
+import json 
+
+class RestClient:
+
+    get_cmds = {"QUOTE":1, "DUMPLOG":1, "DISPLAY_SUMMARY":1}
+
+    symbol_cmds = {
+        "QUOTE":1,
+        "BUY":1,
+        "SELL":1, 
+        "SET_BUY_AMOUNT":1, 
+        "CANCEL_SET_BUY":1, 
+        "SET_BUY_TRIGGER":1, 
+        "SET_SELL_AMOUNT":1, 
+        "SET_SELL_TRIGGER":1, 
+        "CANCEL_SET_SELL":1
+    }
+
+    amount_cmds = {
+    "ADD":1,
+    "BUY":1,
+    "SELL":1,
+    "SET_BUY_AMOUNT":1,
+    "SET_BUY_TRIGGER":1,
+    "SET_SELL_AMOUNT":1,
+    "SET_SELL_TRIGGER":1,
+    }
+
+    def __init__(self, endpoint):
+        self.endpoint = endpoint
+
+    def send_cmd(self, user_id, cmd):
+
+        cmd_type, req_data = self.build_request(user_id, *cmd.split(" "))
+        url = self.build_url(self.endpoint, cmd_type)
+        
+        r = requests.get(url, data=json.dumps(req_data)) if cmd_type in  self.get_cmds else requests.post(url, data=json.dumps(req_data))
+
+    def build_url(self, endpoint, cmd_type):
+        url = "{}{}{}".format(endpoint, "/", cmd_type.lower())
+        return url
+
+    def build_request(self, user_id, *args):
+        data = {
+            "user_id": None, 
+            "symbol": None,
+            "amount": None,
+            "filename": None
+        }
+
+        cmd_type =  args[0]
+
+        data["user_id"] = user_id if user_id != "adminxxx" else None
+        
+        if cmd_type == "DUMPLOG":
+            data["filename"] = args[2] if len(args) > 2 else args[1]
+
+        if cmd_type in  self.symbol_cmds:
+            data["symbol"] = args[2] if len(args) > 2 else None
+        
+        if cmd_type in  self.amount_cmds:
+            if cmd_type == "ADD":
+                data["amount"] = float(args[2]) if len(args) > 2 else None
```

Comment:
For amounts on the backend, we use decimals. Let's stay consistent.

Reply:
So after a bit of research i realize that a Decimal object is not JSON serializable. So to serialize it I have to write my own function which just essentially turns it into a float or string to serialize. I can do that if you like, but it amounts to the same think IMO.

Reply:
Oh ok gotcha. Yeah float should be fine then.